tests/GoogleAuthenticatorTest.php

```diff
@@ -84,13 +121,52 @@ public static function testCheckCodeData(): array
             [true, '2012-03-17 22:17:00'],
             [true, '2012-03-17 22:17:30'],
             [false, '2012-03-17 22:18:00'],
-            [false, 'this date cannot be resolved and results into false'],
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    public static function testCheckCodeDiscrepancyData(): array
+    {
+        return [
+            [false, '2012-03-17 22:16:59'],
+            [true, '2012-03-17 22:17:00'],
+            [true, '2012-03-17 22:17:29'],
+            [false, '2012-03-17 22:17:30'],
+        ];
```

Comment:
If you could find a description for each case, that would be great:
```
return [
    '1second before the interval or whatever' => [false, '2012-03-17 22:16:59'],
];
```
No big deal if you can't
```